src/main/kotlin/no/nav/faktureringskomponenten/service/FakturaserieService.kt

```diff
@@ -61,25 +62,26 @@ class FakturaserieService(
 
         check(opprinneligFakturaserie.erAktiv()) { "Bare aktiv fakturaserie kan erstattes" }
 
-        val avregningsfaktura = avregningBehandler.lagAvregningsfakturaer(
+        val avregningsfakturaer = avregningBehandler.lagAvregningsfakturaer(
             fakturaserieDto.perioder,
             opprinneligFakturaserie.bestilteFakturaer()
         )
         // FIXME: Dette er en midlertidig løsning for å fikse https://jira.adeo.no/browse/MELOSYS-6957
-        val fakturerbarePerioder = PeriodiseringUtil.delIFakturerbarePerioder(fakturaserieDto.perioder, fakturaserieDto.intervall)
+        val fakturerbarePerioderPerIntervall = PeriodiseringUtil.delIFakturerbarePerioder(fakturaserieDto.perioder, fakturaserieDto.intervall)
 
-        val nyeFakturaPerioder = fakturerbarePerioder.filter { faktuerbarPeriode ->
-            avregningsfaktura.none { faktuerbarPeriode.overlaps(LocalDateRange.of(it.getPeriodeFra(), it.getPeriodeTil())) }
+        val nyeFakturaPerioder = fakturerbarePerioderPerIntervall.flatMap { periode ->
+            if (avregningsfakturaer.none { periode.overlaps(LocalDateRange.of(it.getPeriodeFra(), it.getPeriodeTil())) }) listOf(periode)
+            else avregningsfakturaer.map { LocalDateRange.of(it.getPeriodeFra(), it.getPeriodeTil()) }.filter { it.overlaps(periode) && !it.encloses(periode) }.flatMap { periode.substract(it) }
         }
-        val nyFaktura: List<Faktura> = nyeFakturaPerioder.map {
+        val nyeFakturaerForNyePerioder: List<Faktura> = nyeFakturaPerioder.map {
```

Comment:
Dette ble rimelig komplisert kode. 
Tenker det vil hjelpe litt å flytte dette ut til en egen metode. Prøvde dette hos meg og blir da slik
```kotlin
    private fun lagNyeFakturaerForNyePerioder(
        fakturaseriePerioder: List<FakturaseriePeriode>,
        fakturaserieIntervall: FakturaserieIntervall,
        avregningsfakturaer: List<Faktura>,
    ): List<Faktura> {
        val fakturerbarePerioderPerIntervall = PeriodiseringUtil.delIFakturerbarePerioder(fakturaseriePerioder, fakturaserieIntervall)

        val nyeFakturaPerioder = fakturerbarePerioderPerIntervall.flatMap { periode ->
            if (avregningsfakturaer.none { periode.overlaps(LocalDateRange.of(it.getPeriodeFra(), it.getPeriodeTil())) }) listOf(periode)
            else avregningsfakturaer.map { LocalDateRange.of(it.getPeriodeFra(), it.getPeriodeTil()) }
                .filter { it.overlaps(periode) && !it.encloses(periode) }.flatMap { periode.substract(it) }
        }
        val nyeFakturaerForNyePerioder: List<Faktura> = nyeFakturaPerioder.map {
            val perioder = fakturaseriePerioder.filter { periode -> LocalDateRange.of(periode.startDato, periode.sluttDato).overlaps(it) }
            fakturaGenerator.lagFaktura(it.start, it.end, perioder)
        }
        return nyeFakturaerForNyePerioder
    }

```
Tenker også at denne metoden burde ha egne tester som også vil hjelpe til å forklare hva som skjer.

Reply:
Akkurat denne metoden har jeg ikke endret, men jeg er enig i at koden har blitt mer komplisert.
Tilstanden anser jeg fortsatt som det står her:
>        // FIXME: Dette er en midlertidig løsning for å fikse https://jira.adeo.no/browse/MELOSYS-6957

Vi får prod til å fungere snarest mulig og tanken er at jeg skall jobbe med refaktorering i https://jira.adeo.no/browse/MELOSYS-6980

Reply:
Jeg er enig at det er lurt å flytte koden i egen metode for bedre lesbarhet.
Synes imidlertid at testing på komptestnivå er grei for denne PR-en